src/logger.ts

```diff
@@ -0,0 +1,25 @@
+type LogFunction = (...args: unknown[]) => void
+
+const noopLogger: LogFunction = () => {
+  // no-op
+}
+
+interface Logger {
+  system: LogFunction
+  user: LogFunction
+}
+
+const getLogger = (systemLogger?: LogFunction, debug = false): Logger => {
+  // If there is a system logger configured, we'll use that. If there isn't,
+  // we'll pipe system logs to stdout if `debug` is enabled and swallow them
+  // otherwise.
+  const system = systemLogger ?? (debug ? console.log : noopLogger)
+
+  return {
+    system,
+    user: console.log,
```

Comment:
We do not use that anywhere yet?

Reply:
We don't. I thought it'd be nice to have it here for when we do need it, but I can also remove it if you prefer.

Reply:
No all fine just wondering.